src/v/wasm/wasmtime.cc

```diff
@@ -156,9 +156,12 @@ class memory : public ffi::memory {
       : _ctx(ctx)
       , _underlying() {}
 
-    void* translate_raw(size_t guest_ptr, size_t len) final {
-        auto memory_size = wasmtime_memory_data_size(_ctx, &_underlying);
-        if ((guest_ptr + len) > memory_size) [[unlikely]] {
+    void* translate_raw(ffi::ptr guest_ptr, uint32_t len) final {
+        size_t memory_size = wasmtime_memory_data_size(_ctx, &_underlying);
+        // Prevent overflow by upgrading to a larger type.
+        size_t end_read_addr = guest_ptr();
```

Comment:
>`guest_ptr()`

I didn't realize `named_type` had this much machinery inside it 🤦

Reply:
Are you referring to the call operator could access the underlying value?

Reply:
Yeah I originally had a comment here to the effect of "wth is this syntax", then I went and looked at the underlying type.

Reply:
Yeah personally I would have made an explicit `.value()` method instead of the call operator, but 🤷